### PR TITLE
Updated to snake cases, solved bugs and Enhancement

### PR DESCRIPTION
**Pull request by MUTHU-RAKESH-27**
Network Settings Intent - Fixed issue with the message of the day key in the code, fixed the SNMP and Syslog servers updation issue, and Removed the servers.
Device Credential Intent - Camel Case to Snake Case, fixed the bug for throwing errors for multiple deletions
Template Intent - Camel Case to Snake Case, Skeleton for Export and Import Project/Template.
Playbooks - Updated Playbooks for all the above modules.
dnac.py - Updated check_task_response_status() function

### CODE REVIEW

File: playbooks/network_settings_intent.yml [hunk 1]
```diff
@@ -19,68 +19,89 @@
...
+        dnac_password: "{{ dnac_password }}"
+        dnac_verify: "{{ dnac_verify }}"
+        dnac_debug: "{{ dnac_debug }}"
+        dnac_log: True
+        state: deleted
+        config:
+        - global_pool_details:
+            settings:
+              ippool:
+              - ip_pool_name: Global_Pool2
```
Comment (madhansansel) [inline playbooks/network_settings_intent.yml:102]: this as simply name instead of ip_pool_name?

File: playbooks/device_credential_intent.yml [hunk 2]
```diff
@@ -19,67 +19,95 @@
...
-            snmpV3Id: d8974823-250a-41b0-8c9b-b27b2ae01472
-            # httpReadDescription:
-            # httpReadUsername:
-            httpRead: d5d7af00-5a38-4ac1-9f55-03338d00c415
-            # httpWriteDescription:
-            # httpWriteUsername:
-            httpWrite: bec9818e-30cd-468b-bf75-292beefc2e20
-            siteName:
+          assign_credentials_to_site:
+            # cli_description:
```
Comment (madhansansel) [inline playbooks/device_credential_intent.yml:68]: Can't we split this into
```
assign_credentials_to_site:
cli:
    description:
    username:
    id:
snmp_v2:
    read_description:
    write_description:
    write_id:
http:
    read_description:
```

File: playbooks/network_settings_intent.yml [hunk 3]
```diff
@@ -19,68 +19,89 @@
         dnac_log: True
         state: merged
         config:
-        - GlobalPoolDetails:
+        - global_pool_details:
             settings:
               ippool:
-              - ipPoolName: Global_Pool2
+              - ip_pool_name: Global_Pool2
```
Comment (madhansansel) [inline playbooks/network_settings_intent.yml:25]: we can use name instead of ip_pool_name.

File: playbooks/network_settings_intent.yml [hunk 4]
```diff
@@ -19,68 +19,89 @@
         dnac_log: True
         state: merged
         config:
-        - GlobalPoolDetails:
+        - global_pool_details:
             settings:
               ippool:
```
Comment (madhansansel) [inline playbooks/network_settings_intent.yml:24]: can we use ip_pool

File: plugins/modules/template_intent.py [hunk 5]
```diff
@@ -250,297 +513,590 @@
...
+                description: Configuration Template Import Template's payload.
+                elements: dict
+                suboptions:
+                  author:
+                    description: Author of template.
+                    type: str
+                  composite:
+                    description: Is it composite template.
+                    type: bool
+                  containingTemplates:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:609]: Please align with our snake case.

File: plugins/modules/template_intent.py [hunk 6]
```diff
@@ -1273,134 +2282,260 @@ def validate_input_merge(self, template_exists):
...
+        """
+        Update/Create templates and projects in DNAC with fields provided in DNAC.
+
+        Parameters:
+            config (dict) - Playbook details containing template information.
+
+        Returns:
+            self
+        """
+
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:2431]: Can we make it as 3 functions (update_configuration_templates,  .handle_export, handle_import for better readability and maintain the code?
```
def get_diff_merged(self, config):
    self.update_configuration_templates(config) 
    if (self.status == "failed") {
         return self; # we can collect the return status in caller and do needed things.
    }

    self.handle_export(config)
    if (self.status == "failed") {
         return self; # we can collect the return status in caller and do needed things.
    }

    self.handle_import(config)
    if (self.status == "failed") {
         return self; # we can collect the return status in caller and do needed things.
    }
    self.msg = "Successfully completed merged state execution"
    self.status = "success"
    return self
}
```

and collect the return using check_return_status in caller?

File: plugins/modules/template_intent.py [hunk 7]
```diff
@@ -1273,134 +2282,260 @@ def validate_input_merge(self, template_exists):
...
+                if project_created:
+                    self.log("project created with projectId : {0}".format(project_id))
+                else:
+                    self.status = "failed"
+                    self.msg = "Project creation failed"
+                    return self
+
+            is_template_found = self.have_template.get("template_found")
+            template_params = self.want.get("template_params")
+            self.log(str(template_params))
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:2447]: self.log(f"Template Params: {template_params}")
self.log(f"Have Template: {self.have_template}")

File: plugins/modules/template_intent.py [hunk 8]
```diff
@@ -1273,134 +2282,260 @@ def validate_input_merge(self, template_exists):
...
+            self.log(str(self.have_template))
+            template_id = None
+            template_updated = False
+            self.validate_input_merge(is_template_found).check_return_status()
+            if is_template_found:
+                if self.requires_update():
+                    template_id = self.have_template.get("id")
+                    template_params.update({"id": template_id})
+                    self.log(str(self.have_template))
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:2456]: self.log(f"Updating Existing Template: {self.have_template}")

File: plugins/modules/template_intent.py [hunk 9]
```diff
@@ -1273,134 +2282,260 @@ def validate_input_merge(self, template_exists):
...
+                        family="configuration_templates",
+                        function="update_template",
+                        params=template_params,
+                        op_modifies=True,
+                    )
+                    template_updated = True
+                    self.log("Updating Existing Template")
+                else:
+                    # Template does not need update
+                    self.result['response'] = self.have_template.get("template")
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:2468]: can we simply write like below?
```
        self.result.update({
            'response': self.have_template.get("template"),
            'msg': "Template does not need update",
            'status': "exited"
        })
```

File: plugins/modules/template_intent.py [hunk 10]
```diff
@@ -1273,134 +2282,260 @@ def validate_input_merge(self, template_exists):
...
+                    "comments": self.want.get("comments"),
+                    "templateId": template_id
+                }
+                response = self.dnac_apply['exec'](
+                    family="configuration_templates",
+                    function="version_template",
+                    op_modifies=True,
+                    params=version_params
+                )
+                task_details = {}
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:2492]: not required

File: plugins/modules/template_intent.py [hunk 11]
```diff
@@ -1273,134 +2282,260 @@ def validate_input_merge(self, template_exists):
...
+                self.result['diff'] = config.get("configuration_templates")
+                self.log(str(task_details))
+                self.result['response'] = task_details if task_details else response
+
+                if not self.result.get('msg'):
+                    self.msg = "Error while versioning the template"
+                    self.status = "failed"
+                    return self
+
+        export = config.get("export")
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:2510]:
```
def handle_export(self, config):
    export = config.get("export")
    if export:
        self.export_project_or_template(export)
    return self
```

File: plugins/modules/template_intent.py (hunk 7 above)
Comment (MUTHU-RAKESH-27) [reply plugins/modules/template_intent.py:2447]: f string are only supported aboce 3.1 python version 
so we will get error in sanity

File: plugins/modules/template_intent.py (hunk 8 above)
Comment (MUTHU-RAKESH-27) [reply plugins/modules/template_intent.py:2456]: f string are only supported aboce 3.1 python version 
so we will get error in sanity

**Pull request merged**